python/essex-config/essex_config/utils.py

```diff
@@ -0,0 +1,9 @@
+"""Essex Config utils."""
+
+from pydantic import BaseModel
+from pydantic._internal._utils import lenient_issubclass
```

Comment:
Using code that's meant to be internal seems unsafe

Reply:
I agree, but I think it's the cleanest way to tackle this in the short term. Hopefully when Synapse and other corp. environments move to 3.11 we can remove this altogether. 

In the meantime, that's the main reason why I isolated the import to this utils file.

Reply:
Just curious... issubclass should work on 3.10, and pydantic's code uses it. What's the issue you were facing?

Reply:
It throws at runtime because BaseModel is an ABC, causing tests to fail

Reply:
Related issues are: 
https://github.com/pydantic/pydantic/issues/545
https://github.com/pydantic/pydantic/discussions/5970

Reply:
I can think of a way to avoid using this internal function...

Maybe we could go back to the decorator but only use it to add an internal class variable named `__essex_config_enabled__ = True` and then we just use getattr() to the class and check if it's `True`. The downside of this is that we would need to add the decorator to every class used in the config and inner classes. 

I think it's ok to use the internal pydantic function until we can use python 3.11 since it's something we can't control from our side.

Reply:
I'd prefer not to have decorators everywhere.  That limits our ability to compose config stacks out of lower-level models that may be domain-specific.